Revisión Maintenance collaboration.es.rmd es auto (#597)
ropensci/dev_guide: maintenance_collaboration.es.Rmd

# Guía de colaboración {#collaboration}

```{block, type='summaryblock'}
Colaborar con otras personas mejorará tu paquete, y si les das a algunas de ellas rol de autoría y [permisos de escritura en el repositorio](https://help.github.com/articles/repository-permission-levels-for-an-organization/), tu paquete se desarrollará de forma más sostenible.
Además, ¡trabajar en equipo puede ser muy agradable!

Los consejos para colaboración de este capítulo están divididos en una sección sobre [cómo hacer que la infraestructura de tu repo sea accessible para la contribución y la colaboración](#friendlyfiles) (código de conducta, directrices de contribución, etiquetas de *issue*) y [una sección sobre cómo colaborar con nuevas personas que quieran contribuir](#workingcollaborators), en particular en el contexto de la organización "ropensci" de rOpenSci en GitHub.

Además de estos consejos, en su mayoría técnicos, es importante recordar que hay que ser amable y tener en cuenta la perspectiva del resto, especialmente cuando sus prioridades difieren de las tuyas.
```

> ## Haz que tu repositorio sea accesible a las contribucinoes y colaboración {#friendlyfiles}

Reviewer: ```suggestion
## Haz que tu repositorio sea accesible a las contribuciones y colaboración {#friendlyfiles}
```

Por favor, añade este texto al *README*

```markdown
Ten en cuenta que este paquete se publica con un [Código de Conducta para contribuciones](https://ropensci.org/code-of-conduct/). 
Al contribuir a este proyecto, te comprometes a cumplir sus condiciones.
```

Y eliminar el código de conducta actual del paquete, si lo hubiera.

### Guía de contribución

Puedes utilizar guías de *issue*, *pull request* y de contribución.
Tener un archivo de contribución como `.github/CONTRIBUTING.md` o `docs/CONTRIBUTING.md` es obligatorio.
Una forma fácil de insertar una plantilla para una guía de contribución es la función [`use_tidy_contributing()` del paquete `usethis`](https://usethis.r-lib.org/reference/tidyverse.html) que inserta [esta plantilla](https://github.com/r-lib/usethis/blob/main/inst/templates/tidy-contributing.md) como `.github/CONTRIBUTING.md`.
Un ejemplo más extenso es [esta plantilla de Peter Desmet](https://gist.github.com/peterdesmet/e90a1b0dc17af6c12daf6e8b2f044e7c) o las completas [páginas wiki de GitHub para el paquete mlr3](https://github.com/mlr-org/mlr3/wiki).

> Estas y otras plantillas generalmente tendrán que ser modificadas para su uso con un paquete rOpenSci, en particular haciendo referencia y enlazando con nuestro [Código de Conducta](https://ropensci.org/code-of-conduct/) como se describe en otra sección [de este libro](#code-of-conduct).

Reviewer: ```suggestion
Estas y otras plantillas generalmente tendrán que ser modificadas para su uso con un paquete rOpenSci, en particular haciendo referencia y enlazando con nuestro [Código de Conducta](https://ropensci.org/codigo-de-conducta/) como se describe en otra sección [de este libro](#code-of-conduct).
```

Consulta [ejemplos de issues para principiantes en todos los repos de ropensci](https://github.com/search?q=user%3Aropensci+user%3Aropenscilabs+label%3ABeginner+state%3Aopen&type=Issues).

#### Resaltado de *issues*

Puedes [resaltar hasta 3 issues por repositorio](https://docs.github.com/en/articles/pinning-an-issue-to-your-repository) que aparecerán en la parte superior de tu gestor de *issues* como bonitas tarjetas.
Puede ayudar a anunciar cuáles son tus prioridades.

#### Hitos

Puedes [crear hitos](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) y asignarles *issues*, lo que ayuda a ver lo que planeas para la próxima versión de tu paquete, por ejemplo.

### Comunicación

Puedes dirigir a quienes usan tu paquete al foro de rOpenSci, si lo supervisas, o habilitar [Discusiones de GitHub](https://docs.github.com/en/discussions) para el repositorio de tu paquete.
Las discusiones de GitHub pueden convertirse en *issues* si fuera necesario (y al revés, los *issues* pueden convertirse en discusiones).

## Trabajar en equipo {#workingcollaborators}

Lo primero lo primero: ¡estate al tanto de tu repositorio de GitHub!

> - no te olvides de [**vigilar tu repositorio de GitHub**](https://docs.github.com/en/github/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github) para que te notifiquen los *issues* o *pull requests* (si trabajas de a épocas, puedes informar de esto en la guía de contribución).

Reviewer: ```suggestion
- no te olvides de [**vigilar tu repositorio de GitHub**](https://docs.github.com/en/github/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github) para que te notifiquen los *issues* o *pull requests* (si trabajas por períodos, puedes informarlo en la guía de contribución).
```

- no olvides hacer *push* de las actualizaciones que tengas localmente.

- desactiva los test que fallan si no puedes arreglarlos, ya que crean ruido en los *pull requests* que pueden desconcertar quienes comienzan a colaborar.

### Incorporación de miembros al equipo

No hay una regla general en rOpenSci sobre cómo debes incorporar a quienes colaboran en tu paquete al equipo.
Deberías aumentar sus derechos en el repositorio a medida que ganes confianza, y definitivamente deberías reconocer las contribuciones (ver [esta sección](#attributions)).

> Puedes pedir a las nuevas personas que se incorporen que hagan PRs (ver la siguiente sección para evaluar un PR localmente, es decir, más allá de las comprobaciones de CI) a dev/main y evaluarlos antes de aceptarlos, y después de un tiempo dejar que hagan *push* a main. También puede que quieras mantener un sistema de revisión de PRs... ¡incluso para ti una vez que tengas miembros de equipo!

Reviewer: ```suggestion
Puedes pedir a las nuevas personas que se incorporen que hagan PRs (ver la siguiente sección para evaluar un PR localmente, es decir, más allá de las comprobaciones de CI) a *dev/main* y evaluarlos antes de aceptarlos, y después de un tiempo dejar que hagan *push* a *main*. También puede que quieras mantener un sistema de revisión de PRs... ¡incluso para ti una vez que tengas miembros de equipo!
```

Jim Hester ofrece un posible modelo de incorporación de miembros de equipo en [su repositorio de `lintr`](https://github.com/jimhester/lintr/issues/318).

Si tu problema es *reclutar* nuevas personas, puedes publicar una convocatoria abierta como la de Jim Hester [en Twitter](https://twitter.com/jimhester_/status/997109466674819074), [GitHub](https://github.com/jimhester/lintr/issues/318) y, como responsable de un paquete de rOpenSci, puedes pedir ayuda en el Slack de rOpenSci y solicitar al equipo de rOpenSci ideas para reclutar ayuda.

### Trabajar con otras personas (y incluyendo tú en el futuro) {#gitflow}

[Las *branches*](https://happygitwithr.com/git-branches.html) son baratas. 
Utilízalas mucho cuando desarrolles funciones, pruebes nuevas ideas o arregles problemas.

> Si sigues el [desarrollo basado en el "tronco"](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development), "agregarás pequeñas y frecuentes actualizaciones" a tu *branch* principal.

Reviewer: Que les parece Desarrollo troncal en vez de "desarrollo basado en el "tronco"" ?

Author: Me gusta más!

> Si sigues el [desarrollo basado en el "tronco"](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development), "agregarás pequeñas y frecuentes actualizaciones" a tu *branch* principal.

Reviewer: ```suggestion
Si sigues el [desarrollo troncal](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development), "agregarás pequeñas y frecuentes actualizaciones" a tu *branch* principal.
```
